src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1659,51 +1659,87 @@ protected override void InvokeDefaultAction(string path)
                             return;
                         }
 
-                        bool attributeFilter = true;
-                        bool switchAttributeFilter = true;
-                        bool filterHidden = false;           // "Hidden" is specified somewhere in the expression
-                        bool switchFilterHidden = false;     // "Hidden" is specified somewhere in the parameters
-
-                        if (null != evaluator)
+                        // Internal test code, run only if one of the
+                        // 'GciEnumerationAction' test hooks are set.
+                        if (InternalTestHooks.GciEnumerationActionDelete)
                         {
-                            attributeFilter = evaluator.Evaluate(filesystemInfo.Attributes);  // expressions
-                            filterHidden = evaluator.ExistsInExpression(FileAttributes.Hidden);
+                            if (filesystemInfo.Name == "c283d143-2116-4809-bf11-4f7d61613f92")
+                            {
+                                var fullName = Path.Combine(directory.FullName, filesystemInfo.Name);
+                                File.Delete(fullName);
```

Comment:
That was a safety choice. What actually triggers the error is invoking `filesystemInfo.Attributes` further down in the code. I wanted to avoid invoking any properties that might involve lazy evaluation, to avoid having the error occur while running test-hook code.

If we can be sure that using `filesystemInfo.FullName` won't cause the file system to be hit I would be happy to use it instead.